db/db_impl.cc

```diff
@@ -3054,5 +3061,68 @@ void DBImpl::WaitForIngestFile() {
   }
 }
 
+Status DBImpl::StartTrace(const TraceOptions& options,
+                          const std::string& trace_filename) {
+  EnvOptions env_options;
+  unique_ptr<WritableFile> trace_file;
+  Status s = env_->NewWritableFile(trace_filename, &trace_file, env_options);
+  if (!s.ok()) {
+    return s;
+  }
+
+  unique_ptr<WritableFileWriter> file_writer;
+  file_writer.reset(new WritableFileWriter(std::move(trace_file), env_options));
+  unique_ptr<TraceWriter> trace_writer;
+  trace_writer.reset(new FileTraceWriter(std::move(file_writer)));
+
+  return StartTrace(options, std::move(trace_writer));
+}
+
+Status DBImpl::StartTrace(const TraceOptions& /* options */,
+                          std::unique_ptr<TraceWriter>&& trace_writer) {
+  InstrumentedMutexLock lock(&trace_mutex_);
+  tracer_.reset(new Tracer(env_, std::move(trace_writer)));
+  return Status::OK();
+}
+
+Status DBImpl::EndTrace(const TraceOptions& /* options */) {
+  InstrumentedMutexLock lock(&trace_mutex_);
+  Status s = tracer_->Close();
+  tracer_.reset();
+  return s;
+}
+
+Status DBImpl::StartReplay(const ReplayOptions& options,
+                           const std::string& trace_filename,
+                           std::vector<ColumnFamilyHandle*>& handles) {
```

Comment:
Looks like the replayer is a relatively independent component. Can we avoid adding StartReplay() from DB but use Replayer::Replay(DB*) instead?